src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -136,6 +139,19 @@ class HttpGoogleCloudStorageDAO(
     }
   }
 
+  override def getWorkspaces(userId: String): Seq[(WorkspaceName, String)] = {
+    val credential = getGroupServiceAccountCredential
+    val directory = new Directory.Builder(httpTransport, jsonFactory, credential).setApplicationName("firecloud:rawls").build()
+    val groupIds = directory.groups().list().setUserKey(userId).execute().getGroups.map(g => g.getEmail.stripPrefix(s"${groupsPrefix}-").stripSuffix(s"@${appsDomain}")).map(groupId => groupId.split("_", 2))
+
+    val workspaceNames = groupIds.map(tuple => WorkspaceName(tuple.head, tuple(1).splitAt(tuple(1).lastIndexOf("-"))._1)).distinct
+
```

Comment:
I feel like there's a cleaner way to do this. At least, we should make a separate ```unapply```-esque function for group names and put it next to ```makeGroupId```